modules/olm-disconnected-operatorhub-disconnected-mode.adoc

```diff
@@ -0,0 +1,289 @@
+Dockerfile:
++
+----
+$ podman build -f custom-registry.Dockerfile \
+    -t <internal_registry_route>/<namespace>/custom-registry <1>
+----
+<1> Tag the image for the internal registry of the disconnected {product-title}
+cluster and any namespace.
+
+. **Push the Operator catalog image to a registry.**
++
+Your new Operator catalog image must be pushed to a registry that the
+{product-title} can access. This can be the internal registry of the cluster
+itself or another registry that the disconnected cluster has network access to,
+such as an on-premise Quay Enterprise registry.
++
+For this example, login and push the image to the internal registry of the
+disconnected {product-title} cluster
++
+----
+$ podman push <internal_registry_route>/<namespace>/custom-registry
+----
+
+. **Create a CatalogSource pointing to the new Operator catalog image.**
+
+.. Save the following to a file, for example `my-operator-catalog.yaml`:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: my-operator-catalog
+  namespace: default
+spec:
+  displayName: My Operator Catalog
+  sourceType: grpc
+  image: <internal_registry_route>/<namespace>/custom-registry:latest
+----
+
+.. Create the CatalogSource resource:
++
+----
+$ oc create -f my-operator-catalog.yaml
+----
+
+.. Verify the CatalogSource and package manifest are created successfully:
++
+----
+# oc get pods -n openshift-marketplace
+NAME READY STATUS RESTARTS AGE
+my-operator-catalog-6njx6 1/1 Running 0 28s
+marketplace-operator-d9f549946-96sgr 1/1 Running 0 26h
+
+# oc get catalogsource -n openshift-marketplace
+NAME DISPLAY TYPE PUBLISHER AGE
+my-operator-catalog My Operator Catalog grpc 5s
+
+# oc get packagemanifest -n openshift-marketplace
+NAME CATALOG AGE
+etcd My Operator Catalog 34s
+----
++
+You should also be able to view them from the *OperatorHub* page in the web
+console.
+
+. **Mirror the images required by the Operators you want to use.**
+
+.. Determine the images defined by the Operator that you are expecting. This
+ example uses the etcd Operator, requiring the `quay.io/coreos/etcd-operator`
+ image.
+
+.. To use mirrored images, you must first create an ImageContentSourcePolicy for
+each image to change the source location of the Operator catalog image. For
+example:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: etcd-operator
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - <internal_registry_route>:5000/coreos/etcd-operator
+    source: quay.io/coreos/etcd-operator
+----
+
+.. Use the `oc image mirror` command from your workstation with Internet access to
+pull the image from the source registry and push to the internal registry
+without being stored locally:
++
+----
+$ oc image mirror quay.io/coreos/etcd-operator \
+    <internal_registry_route>:5000/coreos/etcd-operator
```

Comment:
Parameterize this port?  e.g. we use `<local_registry_host_name>:<local_registry_host_port>` in a number of other places in the enterprise-4.2 branch (CC @kalexand-rh).